static/app/components/contentSliderDiff/index.tsx

```diff
@@ -0,0 +1,269 @@
+   * Useful when we want to track analytics.
+   */
+  onDividerMouseDown?: (e: React.MouseEvent) => void;
+}
+
+/**
+ * Compares the before and after of visual elements using an adjustable slider.
+ * It allows users to dynamically see the "before" and "after" sections by dragging a divider.
+ * The before and after contents are not directly defined here and have to be provided, so it can be very flexible
+ * (e.g. images, replays, etc).
+ */
+export function ContentSliderDiff({
+  beforeHelp,
+  afterHelp,
+  onDividerMouseDown,
+  beforeContent,
+  afterContent,
+  minHeight = '0px',
+}: ContentSliderDiffProps) {
+  const positionedRef = useRef<HTMLDivElement>(null);
+  const viewDimensions = useDimensions({elementRef: positionedRef});
+  const width = toPixels(viewDimensions.width);
+
+  return (
+    <Fragment>
+      <DiffHeader>
+        <BeforeAfter label={t('Before')} help={beforeHelp} />
+        <BeforeAfter label={t('After')} help={afterHelp} />
+      </DiffHeader>
+      <OverflowVisibleContainer>
+        <Positioned style={{minHeight}} ref={positionedRef}>
+          {viewDimensions.width ? (
+            <DiffSides
+              viewDimensions={viewDimensions}
+              width={width}
+              onDividerMouseDown={onDividerMouseDown}
+              beforeContent={beforeContent}
+              afterContent={afterContent}
+            />
+          ) : (
+            <div />
+          )}
+        </Positioned>
+      </OverflowVisibleContainer>
+    </Fragment>
+  );
+}
+
+const BORDER_WIDTH = 3;
+
+interface DiffSidesProps
+  extends Pick<
+    ContentSliderDiffProps,
+    'onDividerMouseDown' | 'beforeContent' | 'afterContent'
+  > {
+  viewDimensions: {height: number; width: number};
+  width: string | undefined;
+}
+
+function DiffSides({
+  onDividerMouseDown,
+  viewDimensions,
+  width,
+  beforeContent,
+  afterContent,
+}: DiffSidesProps) {
+  const beforeElemRef = useRef<HTMLDivElement>(null);
+  const dividerElem = useRef<HTMLDivElement>(null);
+
+  const resizableDrawer = useResizableDrawer({
+    direction: 'left',
+    initialSize: viewDimensions.width / 2,
+    min: 0,
+    onResize: newSize => {
+      const maxWidth = viewDimensions.width - BORDER_WIDTH;
+      if (beforeElemRef.current) {
+        beforeElemRef.current.style.width =
+          viewDimensions.width === 0
+            ? '100%'
+            : (toPixels(Math.max(BORDER_WIDTH, Math.min(maxWidth, newSize))) ?? '0px');
+      }
+      if (dividerElem.current) {
+        dividerElem.current.style.left =
+          toPixels(Math.max(BORDER_WIDTH, Math.min(maxWidth, newSize))) ?? '0px';
+      }
+    },
+  });
+
+  const handleDividerMouseDown: React.MouseEventHandler<HTMLElement> = useCallback(
+    (event: React.MouseEvent<HTMLElement>) => {
+      onDividerMouseDown?.(event);
+      resizableDrawer.onMouseDown(event);
+    },
+    [onDividerMouseDown, resizableDrawer]
```

Comment:
note: there’s no need to wrap this function in `useCallback`. All we do is pass it to a functional component as a prop.

also, we have `resizableDrawer` as part of the dependency array here, which comes from `useResizableDrawer`, which returns a non-memoized object:

https://github.com/getsentry/sentry/blob/2f5ae60d8252fb5b418da086442aea7a4600f17b/static/app/utils/useResizableDrawer.tsx#L172

so this `useCallback` would be re-computed on every render anyways.